src/SrBot.ts

```diff
@@ -59,38 +61,43 @@ export class SrBot {
                 message.channel.send('Sorry, this server has no players associated.');
             } else {
                 const players = requestedServer.players;
-                let text = this.buildSRTextList(players);
+                let embed = new Discord.RichEmbed()
+                    .setTitle(`Leaderboard: ${requestedServer.teamName}`)
+                    .setColor(HEX_EMBED_COLOR)
+                    .setFooter(`Last updated: ${new Date(this.statsGenerator.getLastUpdate())}`)
+                    .setAuthor('SR Bot', undefined, 'https://github.com/Derekholio/sr-bot');
+                embed = this.appendPlayersToEmbed(players, embed);
```

Comment:
Initializing `embed` to A then setting to B seems a bit dodgy.  I reckon it should be const, then unfactor your loops in `appendPlayersToEmbed` to be run here using `embed.addField(...)` to add your lines.

It may be worth just adding a sort method or util to return you provided players sorted by SR, calling that, then run the forEach to add embed lines.

```
const embed = new Discord(....);
sortPlayersBySr(players).forEach(player => {
    embed.addField(player.player, `${player.SR}${player.private ? ' [PRIVATE]' : ''} ${getRankEmoji(player.SR)}`);
});
```

Reply:
The problem here is that the Discord.RichEmbed is a message builder. Each call to setAuthor, addField, etc returns a new Discord.RichEmbed, rather than appending to the existing one.

Reply:
I was wrong, it does not work that way ;) We should be able to do this no problem